### PR TITLE
Portal gun fix + portal qol

### PR DESCRIPTION
**Pull request by Ebb-Real**
## About The Pull Request

If a portal doesn't teleport you anywhere it will be impassable and can be used as an indestructible wall. That turns the portal gun into a wall gun as long as you don't create the second portal which is not intended. This is fixed in this PR by creating a portal subtype for player created portals, the player created portals can be passed if they aren't linked to anything and can also be crawled under. The reason for a new subtype is to not mess up all the other portal subtypes that don't need to be linked to function or shouldn't be crawled under.

## Why It's Good For The Game

No indestructible wall guns and you can now crawl under player made portals if you don't wanna teleport but have to pass.

## Changelog

:cl:
qol: You can now crawl under player-made portals
fix: Unlinked portals from the portal gun can no longer be used as indestructible walls, and you can now pass them if it isn't linked to another portal.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -140,6 +140,17 @@
...
+/obj/effect/portal/projected
+	name = "projected portal"
+	desc = "A manipulated portal projected by some kind of techknowledgy. It looks like you could crawl under it."
+
+/obj/effect/portal/projected/CanAllowThrough(mob/living/mover, border_dir)
+	if(!linked)
+		return TRUE
+	if(mover.resting & TRUE)
+		return TRUE
+	..()
```
Comment (ZephyrTFA) [inline code/game/objects/effects/portals.dm:152]:
```suggestion
	return ..()
```

**Comment by Ebb-Real** [comment]
i wanna change this around a bit

**Comment by Ebb-Real** [comment]
im unsure why making the `nonlinked_pass` and `resting_pass` don't work when set to false on another subtype

**Comment by anclandor** [comment]
I would rather have the chance of getting cut off by a portal than not be able to go through portals when knocked down.  I feel like its a massive nerf to the utility of the hand tele if you can't enter it well resting or knocked down.

**Comment by Ebb-Real** [comment]
> I would rather have the chance of getting cut off by a portal than not be able to go through portals when knocked down. I feel like its a massive nerf to the utility of the hand tele if you can't enter it well resting or knocked down.

you can still enter it while knocked down by clicking the portal while adjacent to it

**Comment by Ebb-Real** [comment]
> > I would rather have the chance of getting cut off by a portal than not be able to go through portals when knocked down. I feel like its a massive nerf to the utility of the hand tele if you can't enter it well resting or knocked down.
> 
> you can still enter it while knocked down by clicking the portal if you're adjacent to it or under it

forgot to mention you should still be able to shove people into it and enter it normally if you are knocked down since it dosen't count as resting

i still need to fix this though


> im unsure why making the `nonlinked_pass` and `resting_pass` don't work when set to false on another subtype of portal

File: code/game/objects/effects/portals.dm [hunk 2]
```diff
@@ -53,9 +48,13 @@
 		return TRUE
 
 /obj/effect/portal/CanAllowThrough(atom/movable/mover, border_dir)
-	. = ..()
+	var/mob/living/living_mover = mover
```
Comment (ZephyrTFA) [inline code/game/objects/effects/portals.dm:52]:
```suggestion
	if(!..())
		return FALSE
	var/mob/living/living_mover = mover
```

**Comment by Ebb-Real** [comment]
it stops the passing entirely

**Comment by Kylerace** [comment]
yo we have too many prs and this is a fix that has been up for almost a month. did you mean to keep it drafted for 17 days or no @Ebb-Real

**Comment by Ebb-Real** [comment]
ah, yeah i can just close this and come back to it if i figure out what's wrong with the vars. Been a bit busy with fixing issues that my bandana PR caused and studying.

**Pull request closed without merging**